doc/contributing/py-2-3-guide.rst

```diff
@@ -0,0 +1,117 @@
+=====================
+CKAN Python 2/3 Guide
+=====================
+
+This section documents the requirements for porting existing CKAN code for Python 3 support, and for creating new code that supports both Python 2 and Python 3.
+
+----------
+User story
+----------
+
+As a developer working on CKAN or CKAN Extensions, I want a clear guide, for writing code compatible with Python 2 and Python 3, so my code is consistent with common practice around CKAN, using a common set of tools.
+
+---------
+Use cases
+---------
+
+- Contributing Python 2/3 compatible code in CKAN Core
+- Contributing Python 2/3 compatible code in a CKAN Extension
+- Starting a new CKAN Extension with Python 2/3 compatibility
+
+--------
+Python 3
+--------
+
+From CKAN 2.9, CKAN will officially support Python 2 (≥ 2.6) and Python 3 (≥ 3.6).
+
+-----
+Flask
+-----
+
+CKAN provides Python 3 support by removing the underlying Pylons web framework, which is only compatible with Python 2, and integrating Flask as the web framework, which is compatible with Python 2 and Python 3. Flask is a very popular framework for writing web applications in Python; read more about Flask_.
+
+--------
+Approach
+--------
+
+CKAN provides support for both Python 2 and Python 3 using a compatible source approach. This enables us to continue supporting Python 2 in the current transitional period, and gracefully upgrade to a Python 3 only codebase in future versions of CKAN (from CKAN 3.0 onwards).
+
+----
+Tips
+----
+
+Future imports
+##############
+
+Import the following at the top of all modules::
+
+  from __future__ import absolute_import
+  from __future__ import division
+  from __future__ import print_function
+  from __future__ import unicode_literals
```

Comment:
:+1: from me

Reply:
Shouldn't we add only the ones needed? I guess it makes sense `unicode_literals` is always present but for instance the print function is only needed in the cli module. I only could find divisions on 5 modules. And regarding the imports, the current codebase favours absolute imports anyway so perhaps we should recommend that and not load `absolute_import`